assets/src/edit-story/components/statusCheck/statusCheck.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect } from 'react';
+import { useFeatures } from 'flagged';
+/**
+ * Internal dependencies
+ */
+import { useAPI, useConfig, useStory } from '../../app';
+import getStoryMarkup from '../../output/utils/getStoryMarkup';
+
+function StatusCheck() {
+  const {
+    actions: { getStatusCheck },
+  } = useAPI();
+
+  const { metadata } = useConfig();
+  const flags = useFeatures();
+  const { statusCheck } = flags;
+
+  const { story, pages } = useStory((state) => ({
+    story: state.state.story,
+    pages: state.state.pages,
+  }));
+
+  const doStatusCheck = useCallback(() => {
+    if (statusCheck && story && story.status && pages) {
+      const content = getStoryMarkup(story, pages, metadata, flags);
```

Comment:
Changing this to generate real markup as meant that I had to move this lower in app tree. Meaning it is running more often and we have to wait for this pieces of state to be setup. Instead of dong this, should we just mock a html string?

Reply:
Makes sense. Can you ideally use `getStoryMarkup()` on a mock story object? No need to get `story` from state.

Something simple like this should be enough:

```js
getStoryMarkup(
  {
    title: 'Status Check'
    link: 'https://example.com/',
  },
  [],
  {
    publisher: {}
  }
);
```

Reply:
But your point above, is that we need to send real data here. Is a mock object good enough here?

Reply:
A mock object is good enough.

`getStoryMarkup()` gets us good enough real data, a full HTML document.

Reply:
I now mock the story data. We cab improve this later.